dvc/remote/http.py

```diff
@@ -121,18 +123,53 @@ def _etag(self, url):
 
         return etag
 
-    def _download_to(self, url, file, callback=None):
-        r = self._request("GET", url, stream=True)
+    def _download_to(self, url, target_file, callback=None, resume=False):
+        request = self._request("GET", url, stream=True)
+        partial_file = target_file + ".part"
 
-        with open(file, "wb") as fd:
-            bytes_transfered = 0
+        mode, transferred_bytes = self.__determine_mode_get_transferred_bytes(
+            partial_file, resume
+        )
+
+        self.__validate_present_file_size(transferred_bytes, target_file)
+
+        self.__write_request_content(
+            mode, partial_file, request, transferred_bytes, callback
+        )
+
+        os.rename(partial_file, target_file)
 
-            for chunk in r.iter_content(chunk_size=self.CHUNK_SIZE):
-                fd.write(chunk)
+    def __write_request_content(
```

Comment:
minor: why do we use __ (double underscore) instead of _ (single underscore) for private methods in some places?

Reply:
There is no reason for that, ill fix that.